File: Backend/htbp_parent/service/service_hosp/src/main/java/com/seclib/htbp/hosp/service/ScheduleService.java
package com.seclib.htbp.hosp.service;

import com.seclib.htbp.model.hosp.Schedule;
import com.seclib.htbp.vo.hosp.BookingScheduleRuleVo;
import com.seclib.htbp.vo.hosp.ScheduleOrderVo;
import com.seclib.htbp.vo.hosp.ScheduleQueryVo;
import org.springframework.data.domain.Page;

import java.util.List;
import java.util.Map;

public interface ScheduleService {
    void save(Map<String, Object> paramMap);
    Page<Schedule> selectPage(Integer page, Integer limit, ScheduleQueryVo scheduleQueryVo);

    void remove(String hoscode, String hosScheduleId);

    Map<String, Object> getScheduleRule(Long page, Long limit, String hoscode, String depcode);

    List<Schedule> getScheduleDetail(String hoscode, String depcode, String workDate);

    Map<String,Object> getBookingScheduleRule(Integer page, Integer limit, String hoscode, String depcode);

    Schedule getById(String id);

    //根据排班id获取预约下单数据
    ScheduleOrderVo getScheduleOrderVo(String scheduleId);

    //for mq
    void update(Schedule schedule);
}

File: Backend/htbp_parent/service/service_hosp/src/main/java/com/seclib/htbp/hosp/service/ScheduleService.kt
package com.seclib.htbp.hosp.service


import com.seclib.htbp.model.hosp.Schedule
import com.seclib.htbp.vo.hosp.ScheduleQueryVo
import com.seclib.htbp.vo.hosp.ScheduleOrderVo
import org.springframework.data.domain.Page

interface ScheduleService {
    fun save(paramMap: MutableMap<String, Any?>?)
    fun selectPage(page: Int?, limit: Int?, scheduleQueryVo: ScheduleQueryVo?): Page<Schedule?>?
    fun remove(hoscode: String?, hosScheduleId: String?)
    fun getScheduleRule(page: Long?, limit: Long?, hoscode: String?, depcode: String?): Map<String?, Any?>?
    fun getScheduleDetail(hoscode: String?, depcode: String?, workDate: String?): List<Schedule?>?
    fun getBookingScheduleRule(page: Int?, limit: Int?, hoscode: String?, depcode: String?): Map<String?, Any?>?
    fun getById(id: String?): Schedule?

    //根据排班id获取预约下单数据
    fun getScheduleOrderVo(scheduleId: String?): ScheduleOrderVo?

    //for mq
    fun update(schedule: Schedule?)
}